Intermittent test failures in AdminApiTest.testIncrementPartitionsOfTopic

As seen in https://builds.apache.org/job/pulsar-pull-request/org.apache.pulsar$pulsar-broker/59/testReport/junit/org.apache.pulsar.broker.admin/AdminApiTest/testIncrementPartitionsOfTopic/

```
java.lang.AssertionError: expected [8] but found [12]
	at org.apache.pulsar.broker.admin.AdminApiTest.testIncrementPartitionsOfTopic(AdminApiTest.java:1761)
```

Apparently, this failure is causing other tests in `AdminApiTests` to fail afterwards:

```
org.apache.pulsar.broker.admin.AdminApiTest.testIncrementPartitionsOfTopic
org.apache.pulsar.broker.admin.AdminApiTest.testIncrementPartitionsOfTopic
org.apache.pulsar.broker.admin.AdminApiTest.testNamespaceSplitBundle
org.apache.pulsar.broker.admin.AdminApiTest.testNamespaceUnloadBundle
org.apache.pulsar.broker.admin.AdminApiTest.testPersistentTopicsExpireMessages
org.apache.pulsar.broker.admin.AdminApiTest.testPulsarAdminForUriAndUrlEncoding
org.apache.pulsar.broker.admin.AdminApiTest.testPulsarAdminForUriAndUrlEncoding
```

--- a/pulsar-broker/src/test/java/org/apache/pulsar/broker/admin/AdminApiTest.java
+++ b/pulsar-broker/src/test/java/org/apache/pulsar/broker/admin/AdminApiTest.java
@@ -1734,6 +1734,8 @@ public class AdminApiTest extends MockedPulsarServiceBaseTest {
 
         // (1) update partitions
         admin.persistentTopics().updatePartitionedTopic(partitionedTopicName, newPartitions);
+        // invalidate global-cache to make sure that mock-zk-cache reds fresh data
+        pulsar.getGlobalZkCache().invalidateAll();
         // verify new partitions have been created
         assertEquals(admin.persistentTopics().getPartitionedTopicMetadata(partitionedTopicName).partitions,
                 newPartitions);
@@ -1755,6 +1757,8 @@ public class AdminApiTest extends MockedPulsarServiceBaseTest {
         // newly created partition topics
         consumer2.close();
         consumer2 = client.subscribe(partitionedTopicName, subName2, conf);
+        // sometime: mockZk fails to refresh ml-cache: so, invalidate the cache to get fresh data
+        pulsar.getLocalZkCacheService().managedLedgerListCache().clearTree();
         assertEquals(Sets.newHashSet(admin.persistentTopics().getSubscriptions(newPartitionTopicName)),
                 Sets.newHashSet(subName1, subName2));
 

